apis/r/R/Init.R

```diff
@@ -17,6 +17,39 @@
                        sQuote(rpkg_lib_version), sQuote(soma_lib_version))
         stop(msg, call. = FALSE)
     }
+
+    # This is temporary for https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+    # It will be removed once 2407 is complete.
+    if (Sys.getenv("SOMA_R_NEW_SHAPE") != "") {
+      .pkgenv[["use_current_domain_transitional_internal_only"]] <- TRUE
+      cdmsg <- " SOMA_R_NEW_SHAPE enabled"
+    } else {
+      .pkgenv[["use_current_domain_transitional_internal_only"]] <- FALSE
+      cdmsg <- ""
+    }
+     msg <- sprintf("TileDB Core version %s used by TileDB-R package, but TileDB-SOMA uses %s [%s]",
+                    sQuote(rpkg_lib_version), sQuote(soma_lib_version), sQuote(cdmsg))
+}
+
+# This is temporary only. Please see:
+# * https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+# * https://github.com/single-cell-data/TileDB-SOMA/pull/2950
+.new_shape_feature_flag_enable <- function() {
+    .pkgenv[["use_current_domain_transitional_internal_only"]] <- TRUE
+}
+
+# This is temporary only. Please see:
+# * https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+# * https://github.com/single-cell-data/TileDB-SOMA/pull/2950
+.new_shape_feature_flag_disable <- function() {
+    .pkgenv[["use_current_domain_transitional_internal_only"]] <- FALSE
+}
+
+# This is temporary only. Please see:
+# * https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+# * https://github.com/single-cell-data/TileDB-SOMA/pull/2950
+.new_shape_feature_flag_is_enabled <- function() {
+    .pkgenv[["use_current_domain_transitional_internal_only"]]
```

Comment:
These functions are being defined, but never used (`.new_shape_feature_flag_enabled()` is used in tests, but that's it). Instead, `.pkenv` is being modified (in `.onLoad()`) and accessed (in `get_domain_and_extent_dataframe()` and `get_domain_and_extent_array()`) directly

Reply:
I thought I'd use them for unit-test purposes

It looks like the env-var-driven approach will suffice

I'll take these back out

Reply:
I got rid of the first two; `new_shape_feature_flag_is_enabled` is used in unit tests